js/apps/system/_admin/aardvark/APP/react/src/views/views/arangoSearchView/ArangoSearchJSONEditor.tsx

```diff
@@ -29,9 +32,27 @@ export const ArangoSearchJSONEditor = () => {
         ajv={ajv}
         history
         schema={schema}
+        onChangeText={jsonString => {
+          try {
+            const json = JSON.parse(jsonString);
+            setIsFormDisabled(false);
+            if (JSON.stringify(json) !== JSON.stringify(values)) {
+              setValues(json);
+            }
+          } catch (e) {
+            setIsFormDisabled(true);
+          }
+        }}
         onChange={json => {
-          if (JSON.stringify(json) !== JSON.stringify(values)) {
-            setValues(json);
+          try {
+            const jsonString = JSON.stringify(json);
+            const jsonParsed = JSON.parse(jsonString);
+            setIsFormDisabled(false);
+            if (JSON.stringify(jsonParsed) !== JSON.stringify(values)) {
+              setValues(json);
+            }
+          } catch (e) {
+            setIsFormDisabled(true);
```

Comment:
would we also show an error in this case?

Reply:
do you mean showing an error with `window.arangoHelper.arangoError()`?

Reply:
Errors were shown with `<JSONErrors/>` but I still tried to find a better alternative to my original changes, and I got it, I can control everything from `onValidateError` simpler and more meaningful